src/bin/handle-git-dir-change.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {execSync} from 'child_process';
+import {Changes, FileMode, FileData} from '../types';
+import {logger} from '../logger';
+import {readFile} from 'fs';
+import * as path from 'path';
+
+// data about a file in a git directory
+interface GitFileData {
+  path: string;
+  fileData: FileData;
+}
+
+// See https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-git-diff-filesltpatterngt82308203
```

Comment:
great use of a comment, It's really useful to link the canonical source of magic values like this.